js/src/forum/helpers/hljsDefaults.js

```diff
@@ -393,9 +393,43 @@ export function hljsDefaults(theme, baseUrl) {
 
     var x = document.getElementsByTagName('pre');
     for(var i=0; i<x.length; ++i)
-        x[i].style = 'background: ' + background + ';';
+        x[i].style = 'background: ' + background + ';' + 'color: ' + color;
 
     x = document.getElementsByTagName('code');
     for(var i=0; i<x.length; ++i)
         x[i].style = 'color: ' + color + ';';
+
+    var MutationObserver = window.MutationObserver || window.WebKitMutationObserver || window.MozMutationObserver;
+    var elements = document.querySelectorAll('pre code');
+    var observer = new MutationObserver(function(mutations) {
+        mutations.forEach(function(mutation) {
+            if (mutation.type == "attributes") {
+                show_language();
+            }
+        });
+    });
+
+    elements.forEach(function(element){
+        observer.observe(element, {
+            attributes: true,
+            attributeFilter: ['class']
+        });
+    })
```

Comment:
Asynchronous loading. Add listener to `<pre>` tag, after its class attribute changed,  execute the `show_language` method.